src/main/kotlin/gg/flyte/christmas/lobby/LobbySnowballs.kt

```diff
@@ -0,0 +1,74 @@
+package gg.flyte.christmas.lobby
+
+import gg.flyte.christmas.minigame.world.MapRegion
+import gg.flyte.christmas.minigame.world.MapSinglePoint
+import gg.flyte.christmas.util.eventController
+import gg.flyte.twilight.event.event
+import org.bukkit.GameMode
+import org.bukkit.Material
+import org.bukkit.Sound
+import org.bukkit.entity.Player
+import org.bukkit.event.player.PlayerInteractEvent
+import org.bukkit.inventory.ItemStack
+
+class LobbySnowballs {
+    private val lobbyRegion = MapRegion(
+        MapSinglePoint(545, 100, 500), // PLACEHOLDER VALUES (I DONT HAVE THE REAL MAP)
+        MapSinglePoint(575, 120, 535)
+    )
+
+    init {
+        event<PlayerInteractEvent> {
+            if (!shouldProcessEvent(this)) return@event
+
+            val clickedBlock = clickedBlock ?: return@event
+
+            // Only handle snow-type blocks
+            if (clickedBlock.type != Material.SNOW && clickedBlock.type != Material.SNOW_BLOCK) {
+                return@event
+            }
+
+            // Give snowball and play effects
+            if (player.inventory.firstEmpty() != -1) {
+                giveSnowball(player)
+                playSnowCollectionEffects(player, clickedBlock.location)
+            }
+
+            // Cancel vanilla block breaking
+            isCancelled = true
+        }
+    }
+
+    private fun shouldProcessEvent(event: PlayerInteractEvent): Boolean {
+        return eventController().currentGame == null &&
```

Comment:
This is the only condition that will validate the event. The others are either implicitly known, or never changes (ex. gamemode)

Since this change will reduce this method to one line, I would in-line all its uses.